cmd/mdatagen/internal/metadata.go

```diff
@@ -300,6 +300,8 @@ type Attribute struct {
 	Enum []string `mapstructure:"enum"`
 	// Type is an attribute type.
 	Type ValueType `mapstructure:"type"`
+	// IsTemplateKey defines if attribute template of type Type should be used
+	IsTemplateKey bool `mapstructure:"is_template_key"`
```

Comment:
```suggestion
	// HasTemplateType defines if attribute template of type Type should be used
	HasTemplateType bool `mapstructure:"is_template_type"`
```
Sorry, I think I misunderstood when reading a previous version of this: I think we should call this `has_template_type` or just `is_template` since I don't think "key" has a totally clear meaning in this context. Ultimately we need to find a way to communicate to the user that the type they set above is actually `template[<TYPE>]`.

Alternatively, the `type` option needs to know how to parse generics, but I'm not sure if that's worth the added complexity.

Reply:
adapted, thanks!

Reply:
@dmitryax what would you like to see here? Thinking on it some more, I see the value in aligning with Weaver and supporting the `template[<TYPE>]` syntax, but again I think it may add some additional complexity to the code.

Reply:
repinging here @evan-bradley @dmitryax